src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs

```diff
@@ -5922,27 +5912,122 @@ .maxstack  2
 }");
         }
 
-        [Fact]
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56007")]
         public void TestNullCoalesce_NullableDefault_MissingGetValueOrDefault()
         {
-            var source = @"
-class Program
-{
-    static int Coalesce(int? x)
-    {
-        return x ?? 0;
-    }
-}";
+            var source = """
+                class Program
+                {
+                    static int Coalesce(int? x)
+                    {
+                        return x ?? 0;
+                    }
+                }
+                """;
+
             var comp = CreateCompilation(source);
             comp.MakeMemberMissing(SpecialMember.System_Nullable_T_GetValueOrDefault);
 
+            // We gracefully fallback to calling `GetValueOrDefault(defaultValue)` member
+            comp.VerifyEmitDiagnostics();
+            var verifier = CompileAndVerify(comp);
+
+            verifier.VerifyIL("Program.Coalesce", """
+                {
+                  // Code size        9 (0x9)
+                  .maxstack  2
+                  IL_0000:  ldarga.s   V_0
+                  IL_0002:  ldc.i4.0
+                  IL_0003:  call       "int int?.GetValueOrDefault(int)"
+                  IL_0008:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56007")]
+        public void TestNullCoalesce_NullableDefault_MissingGetValueOrDefaultAndGetValueOrDefaultWithADefaultValueParameter()
+        {
+            var source = """
+                class Program
+                {
+                    static int Coalesce(int? x)
+                    {
+                        return x ?? 0;
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.MakeMemberMissing(SpecialMember.System_Nullable_T_GetValueOrDefault);
+            comp.MakeMemberMissing(SpecialMember.System_Nullable_T_GetValueOrDefaultDefaultValue);
+
             comp.VerifyEmitDiagnostics(
-                // (6,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault'
-                //         return x ?? 0;
-                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "x").WithArguments("System.Nullable`1", "GetValueOrDefault").WithLocation(6, 16),
-                // (6,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault'
+                // (5,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault'
```

Comment:
>// (5,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault' [](http://example.com/codeflow?start=16&length=95)

The error looks surprising. Why can't we simply emit code like in the test below?

Reply:
I see, it also relies on GetValueOrDefault.